File: mvvm/src/main/java/com/base/mvvm/feature/login/LoginViewModel.java
package com.base.mvvm.feature.login;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.view.View;

import com.base.mvvm.bean.BaseBean;
import com.base.mvvm.bean.UserBean;
import com.base.mvvm.databinding.ActivityLoginBinding;
import com.base.mvvm.feature.main.order.list.OrderListActivity;
import com.base.mvvm.util.PrefrenceUtil;

/**
 * 描述: 登陆模块ViewModel实现
 *
 * @author : <a href="mailto:fanhuayong@yinli56.com">Fanhy</a>
 * @version : Ver 1.0
 * @date : 2018-04-18 16:59
 */

public class LoginViewModel {
    private LoginModel model;
    private ActivityLoginBinding binding;

    public LoginViewModel(Activity activity, ActivityLoginBinding binding) {
        this.binding = binding;
        this.model = new LoginModel(activity, this);
    }

    public void login(UserBean userBean) {
        LoginActivity.sIdlingResource.increment();//App 开始进入忙碌状态 ，等待通知
        model.login(userBean);
    }

    public void setLoginResult(BaseBean<Integer> baseBean) {
        if (baseBean.status.equals("success")) {// 登陆成功
            binding.setBaseBean(baseBean);
            UserBean userBean = binding.getUserBean();
            if (userBean.isRecordPwd) {// 保存用户信息
                PrefrenceUtil.setUserBean(userBean);
            } else {// 清除用户信息
                PrefrenceUtil.clearUserBean();
            }
            binding.tvUserWelcome.setVisibility(View.VISIBLE);
            Context context = binding.tvUserWelcome.getContext();
            context.startActivity(new Intent(context, OrderListActivity.class));
        } else {// 登陆失败
            binding.edtUserName.setText("");
            binding.edtUserPwd.setText("");
            binding.tvUserWelcome.setText("");
            binding.tvUserWelcome.setVisibility(View.GONE);
        }
        LoginActivity.sIdlingResource.decrement();//加载完毕后，将App设置成空闲状态
    }
}

File: kotlindemo/src/main/java/com/base/demo/login/LoginViewModel.kt
package com.base.demo.login

import android.content.Intent
import android.widget.Toast
import com.base.demo.bean.BaseBean
import com.base.demo.main.MainActivity

/**
 * 描述: 登陆模块ViewModel层
 *
 * @author : <a href="mailto:fanhuayong@yinli56.com">Fanhy</a>
 * @version : Ver 1.0
 * @date : 2018-05-11 11:17
 */
class LoginViewModel {
    var act: LoginActivity
    var m: LoginModel

    constructor(act: LoginActivity) {
        this.act = act
        m = LoginModel(this)
    }

    fun login(userName: String, userPwd: String) {
        m.login(userName, userPwd)
    }

    fun loginResult(baseBean: BaseBean<Int>) {
        var intent = Intent(act, MainActivity::class.java)
        if ("success" == baseBean.status) {
            intent.putExtra("userBean", act.loginBinding?.userBean)
            act.startActivity(intent)
        } else {
            Toast.makeText(act, "登陆失败！请检查用户名和密码", Toast.LENGTH_SHORT).show()
        }
    }
}